### PR TITLE
add @mesonbuild filetype support

### LINKED ISSUE #1734: Meson filetype support
Files are always called "meson.build"

Here is an example: https://git.gnome.org/browse/glib/plain/meson.build

Here is a configuration for the syntax highlighting library meson uses for their documentation, which might help: https://github.com/mesonbuild/meson/blob/master/docs/theme/extra/prism_components/prism-meson.js

Here is the config file gtksourceview uses: https://git.gnome.org/browse/gtksourceview/tree/data/language-specs/meson.lang

http://mesonbuild.com/

### PR DESCRIPTION
**Pull request by andy5995**
closes #1734

### CODE REVIEW

**Comment by eht16** [comment]
Looks good!
A few questions and remarks:
- `data/filedefs/filetypes.Meson.conf` must be added in `data/Makefile.am` to `filetypes_dist`. Otherwise the config file won't be included in releases and won't be installed
- it'd be nice if we would add the filetype also to the appropriate key in the`[Groups]` section in `data/filetype_extensions.conf` (this is used to place the filetype in the corresponding group in the Document->Set filetype menu) - I'm not sure where it fits best, I guess either in Scripting or Misc
- would you mind sorting the keyword and identifiers list alphabetacally? This makes it easier to update in the future
- could we maybe use `meson.build` in `data/filetype_extensions.conf`? `*.build` might be too greedy, it seems common to me to use `meson.build` but I'm no Meson expert :D

**Comment by elextr** [comment]
> could we maybe use meson.build in data/filetype_extensions.conf? *.build might be too greedy, it seems common to me to use meson.build but I'm no Meson expert :D

AFAIK it can only be meson.build so yeah the full name should be used.

**Comment by andy5995** [comment]
Sure, no problem.  I did all the items you listed @eht16

**Comment by andy5995** [comment]
@eht16 @elextr Should syntax highlighting for meson_options.txt be included as well?

**Comment by elextr** [comment]
There isn't a meson lexer, but if one of the others works well enough it could probably be used.

**Comment by eht16** [comment]
@andy5995 not sure about `meson_options.txt`. I guess compared to `meson.build` it's maybe less often used and the synxtax doesn't seem that complicated so that highlighting is necessary.

The rest looks fine, thanks for the changes.

**Comment by andy5995** [comment]
> @andy5995 not sure about `meson_options.txt`. I guess compared to `meson.build` it's maybe less often used and the synxtax doesn't seem that complicated so that highlighting is necessary.
> 
> The rest looks fine, thanks for the changes.

Sounds like this is ready for approval and merging then

File: data/filetype_extensions.conf [hunk 1]
```diff
@@ -85,5 +86,5 @@ None=*;
 Programming=Arduino;Clojure;CUDA;Cython;Genie;Groovy;Kotlin;Nim;Scala;Swift;
 Script=Graphviz;TypeScript;
 Markup=
-Misc=JSON;
+Misc=JSON;Meson;
```
Comment (elextr) [inline data/filetype_extensions.conf:89]: Cmake and Makefiles are both in the scripting group, so maybe better there.

Comment (andy5995) [reply data/filetype_extensions.conf]: done

**Comment by elextr** [comment]
Looks ok to me, so all it needs is somebody who is not @andy5995 to test it and it can be merged after release.

**Comment by eht16** [comment]
I've tested it (filetype detection, syntax highlighting, make install, make distcheck) and it works fine.

@elextr why do you want it to be merged *after* the release? For me, this could go still into 1.38.

**Comment by elextr** [comment]
I didn't mean it _had_ to be after the release.  I just meant it was _likely_ to be after.

**Comment by eht16** [comment]
Ok, great. So let's merge before 1.38 is getting out :).

**Pull request merged**